### DISCUSSION TITLE
Failed to connect to ESP32

### DISCUSSION THREAD

**Discussion by nmea2000** [hatlabs/discussions#78] [Q&A]
Should I push any buttons ( boot, reset) when uploading to SH-ESP32?

Configuring upload protocol...
AVAILABLE: esp-prog, espota, esptool, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa
CURRENT: upload_protocol = esptool
Looking for upload port...
Auto-detected: /dev/ttyS0
Uploading .pio/build/esp32dev/firmware.bin
esptool.py v3.1
Serial port /dev/ttyS0
Connecting........_____....._____....._____....._____....._____....._____....._____

A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header
*** [upload] Error 2

PlatformIO Core, version 6.1.11

**Comment by nmea2000** [comment]
Holding down the boot button during upload made a diffrens.
Leaving...
Hard resetting via RTS pin...
============================================= [SUCCESS] Took 17.53 seconds =============================================

**Comment by mairas** [comment]
This is a bit surprising - normally the SH-ESP32 does auto-reset at the beginning of the flashing process. But glad you got it working.

**Comment by mairas** [comment]
Re-opening to provide visibility for future readers.
